src/control/controlproxy.cpp

```diff
@@ -3,35 +3,41 @@
 #include "control/controlproxy.h"
 #include "control/control.h"
 
-ControlProxy::ControlProxy(QObject* pParent)
-        : QObject(pParent),
-          m_pControl(NULL) {
+ControlProxy::ControlProxy(const QString& g, const QString& i, QObject* pParent, ControlFlags flags)
+        : ControlProxy(ConfigKey(g, i), pParent, flags) {
 }
 
-ControlProxy::ControlProxy(const QString& g, const QString& i, QObject* pParent)
-        : QObject(pParent) {
-    initialize(ConfigKey(g, i));
+ControlProxy::ControlProxy(const char* g, const char* i, QObject* pParent, ControlFlags flags)
+        : ControlProxy(ConfigKey(g, i), pParent, flags) {
 }
 
-ControlProxy::ControlProxy(const char* g, const char* i, QObject* pParent)
-        : QObject(pParent) {
-    initialize(ConfigKey(g, i));
-}
+ControlProxy::ControlProxy(const ConfigKey& key, QObject* pParent, ControlFlags flags)
+        : QObject(pParent),
+          m_pControl(nullptr) {
+    DEBUG_ASSERT(!key.isNull() || flags.testFlag(ControlFlag::AllowEmptyKey));
+    m_key = key;
 
-ControlProxy::ControlProxy(const ConfigKey& key, QObject* pParent)
-        : QObject(pParent) {
-    initialize(key);
+    if (!flags.testFlag(ControlFlag::InitializeLater)) {
+        flags.setFlag(ControlFlag::InitializeLater, false);
```

Comment:
But we just tested that InitializeLater is **not** set? Why then set it to false again?